class.json-api-endpoints.php

```diff
@@ -1539,16 +1539,28 @@ function current_user_can_access_post_type( $post_type, $context='display' ) {
 
 	function is_post_type_allowed( $post_type ) {
 		// if the post type is empty, that's fine, WordPress will default to post
-		if ( empty( $post_type ) )
+		if ( empty( $post_type ) ) {
 			return true;
+		}
 
 		// allow special 'any' type
-		if ( 'any' == $post_type )
+		if ( 'any' == $post_type ) {
 			return true;
+		}
 
 		// check for allowed types
-		if ( in_array( $post_type, $this->_get_whitelisted_post_types() ) )
+		if ( in_array( $post_type, $this->_get_whitelisted_post_types() ) ) {
 			return true;
+		}
+
+		if ( $post_type_object = get_post_type_object( $post_type ) ) {
+			if ( ! empty( $post_type_object->show_in_rest ) ) {
+				return $post_type_object->show_in_rest;
+			}
+			if ( ! empty( $post_type_object->publicly_queryable ) ) {
```

Comment:
We should double check that the post type object always includes the `publicly_queryable` flag, since the effect we're after is to fall back to the `public` value if `publicly_queryable` is not explicitly set.

>If no value is specified for `exclude_from_search`, `publicly_queryable`, `show_in_nav_menus`, or `show_ui`, they inherit their values from `public`.

https://codex.wordpress.org/Function_Reference/register_post_type